apps/andi/lib/andi_web/live/dataset_live_view.ex

```diff
@@ -22,24 +24,48 @@ defmodule AndiWeb.DatasetLiveView do
           </label>
         </form>
       </div>
-      <%= live_component(@socket, Table, datasets: @datasets) %>
+      <%= live_component(@socket, Table, id: :datasets_table, datasets: @datasets) %>
     </div>
     """
   end
 
   def mount(_session, socket) do
-    {:ok, assign(socket, datasets: nil, search_text: "")}
+    {:ok, assign(socket, datasets: nil, search_text: nil, order: {"data-title", "asc"}, params: %{})}
+  end
+
+  def handle_info({:order, field}, socket) do
+    order_dir =
+      case socket.assigns.order do
+        {^field, "asc"} -> "desc"
```

Comment:
Why do we need to pin the field?

Reply:
this is to allow it to be an explicit pattern match against whatever comes out of the `case` expression. Without the pin, the `field` variable is rebound to whatever comes out of that expression and that first branch will always evaluate to true and execute (the `"desc"` branch).

Reply:
Ah, I was thinking you would just pin if you didn't want field to be changed. I didn't think about how it'd make a more forgiving pattern match for the case statement. Thanks